crates/red_knot_python_semantic/src/types/unpacker.rs

```diff
@@ -268,8 +268,14 @@ pub(crate) struct UnpackResult<'db> {
 }
 
 impl<'db> UnpackResult<'db> {
-    pub(crate) fn get(&self, expr_id: ScopedExpressionId) -> Option<Type<'db>> {
-        self.targets.get(&expr_id).copied()
+    /// Returns the inferred type for a given sub-expression of the left-hand side target
+    /// of an unpacking assignment.
+    ///
+    /// Panics if a scoped expression ID is passed in that does not correspond to a sub-
+    /// expression of the target.
+    #[track_caller]
+    pub(crate) fn type_for(&self, expr_id: ScopedExpressionId) -> Type<'db> {
```

Comment:
We call the same method `expression_type` in `TypeInference`. I'm somewhat undecided if I'd value the consistency here or if diverging is fine

Reply:
Let's go with `expression_type` :+1: